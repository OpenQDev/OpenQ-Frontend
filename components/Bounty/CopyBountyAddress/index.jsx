// Third party
import React from 'react';
import CopyAddressToClipboard from '../../CopyAddressToClipboard';

const CopyBountyAddress = ({ address, styles }) => {
  return (
    <div className={styles || ''}>
      <div className='flex  items-center space-x-2 cursor-pointer'>
        <CopyAddressToClipboard data={address} />
      </div>
    </div>
  );
};

export default CopyBountyAddress;
