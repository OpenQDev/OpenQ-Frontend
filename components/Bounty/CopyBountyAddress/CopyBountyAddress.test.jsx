/**
 * @vi-environment jsdom
 */
import React from 'react';

import CopyBountyAddress from '.';
import Constants from '../../../test-utils/constant';
import renderer from 'react-test-renderer';

describe('CopyBountyAddress', () => {
  const bounty = Constants.bounty;

  beforeEach(() => {
    const observe = vi.fn();
    const disconnect = vi.fn();

    window.IntersectionObserver = vi.fn(() => ({
      observe,
      disconnect,
    }));
  });

  it('should render CopyBountyAddress and match DOM Snapshot', () => {
    const tree = renderer.create(<CopyBountyAddress address={bounty.bountyAddress} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
