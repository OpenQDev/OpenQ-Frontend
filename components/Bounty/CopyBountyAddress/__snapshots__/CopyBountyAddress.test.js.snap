// vi Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyBountyAddress should render CopyBountyAddress and match DOM Snapshot 1`] = `
<div
  className=""
>
  <div
    className="flex  items-center space-x-2 cursor-pointer"
  >
    <div
      className="relative undefined"
    >
      <div
        className="flex flex-row items-center space-x-4 md:space-x-1 cursor-pointer w-fit"
        onClick={[Function]}
      >
        <div
          className="relative flex flex-col items-center group"
        >
          <svg
            className="h-6 w-6"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 16H6a2 2 0 01-2-2V6a2 2 0 012-2h8a2 2 0 012 2v2m-6 12h8a2 2 0 002-2v-8a2 2 0 00-2-2h-8a2 2 0 00-2 2v8a2 2 0 002 2z"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth={2}
            />
          </svg>
        </div>
        <div
          className="px-1"
        >
          0x066efd8702
						...
						f98241d774
        </div>
      </div>
    </div>
  </div>
</div>
`;
